ui/components/app/custom-spending-cap/custom-spending-cap.js

```diff
@@ -127,6 +138,24 @@ export default function CustomSpendingCap({
     }
 
     dispatch(setCustomTokenAmount(String(valueInput)));
+
+    const newData = getCustomTxParamsData(txParams.data, {
+      customPermissionAmount: valueInput,
+      decimals,
+    });
+    const { from, to, value: txValue } = txParams;
+    const estimatedGasLimit = await estimateGas({
+      from,
+      to,
+      value: txValue,
+      data: newData,
+    });
+    if (estimatedGasLimit) {
+      await updateTransaction({
+        gasLimit: hexToDecimal(addHexPrefix(estimatedGasLimit)),
+      });
+    }
```

Comment:
I think it would be good to wrap this in a try/catch. 

If updateTransaction throws an error and it would break here and `setInputChangeInProgress(false)` below would not run 

![Screen Shot 2023-04-13 at 6 54 16 PM](https://user-images.githubusercontent.com/20778143/231891634-e1b716ba-cbed-4b48-8c29-3084294312e7.png)
<img width="742" alt="Screen Shot 2023-04-13 at 6 54 40 PM" src="https://user-images.githubusercontent.com/20778143/231891640-ae4f6fe0-6409-4545-8452-c6ae918831f1.png">

Reply:
yep good point, I was also. thinking of adding try/catch here. Let me do that.